sklearn/cluster/_birch.py

```diff
@@ -457,12 +476,12 @@ def fit(self, X, y=None):
         self
             Fitted estimator.
         """
-        self.fit_, self.partial_fit_ = True, False
-        return self._fit(X)
+        self.__fit, self.__partial_fit = True, False
+        return self._fit(X, partial=False)
```

Comment:
Maybe add a comment "TODO remove in 1.2" so we don't forget (there's another occurence below).